src/main/java/com/mxstrive/jenkins/plugin/contentreplace/ContentReplaceBuilder.java

```diff
@@ -110,9 +112,12 @@ private boolean replaceFileContent(String path, FileContentReplaceConfig config,
 	private boolean replaceFileContent(FilePath filePath, FileContentReplaceConfig config, EnvVars envVars,
 			FilePath workspace, TaskListener listener) throws InterruptedException, IOException {
 		PrintStream log = listener.getLogger();
-		InputStream is = filePath.read();
-		List<String> lines = readLines(is, Charset.forName(config.getFileEncoding()));
-		is.close();
+		Charset charset = Charset.forName(config.getFileEncoding());
+		List<String> lines = Collections.emptyList();
+		try (BufferedInputStream bis = new BufferedInputStream(filePath.read());
+				InputStreamReader isr = new InputStreamReader(bis, charset);) {
+			lines = readLines(isr);
+		}

```

Comment:
Why are you first assigning `List<String> lines` it should be null or ideally not assigned?

Also why isn't there a safe fallback charset, and why is it mandatory? At least use a BOMInputStream to detect if a BOM is present, else the native FS charset should be used.

Reply:
Thank you for your feedback.

The way I see it, it is a good practice to avoid any kind of references to Null in your code. By having an empty list, redundant null checks as well as any possible `NullPointerException` can easily be avoided. Call it force of habit, but feel free to change it.

Regarding your second question: I tried to leave the code semantically intact and merely optimize or fix it (e.g. by using a `BufferedInputStream` explicitly or the try-with-resources statement).